pkg/controller/scylladbcluster/resource.go

```diff
@@ -1391,20 +1406,31 @@ func makeEndpointSliceForLocalIdentityService(sc *scyllav1alpha1.ScyllaDBCluster
 			continue
 		}
 
-		dcEndpoints, err := calculateEndpointsForRemoteDCPods(sc, clientBroadcastAddressType, dc, dcNamespace, dcPodSelector, remotePodLister, remoteServiceLister)
+		dcEndpoints, dcIPFamily, err := calculateEndpointsForRemoteDCPods(sc, clientBroadcastAddressType, dc, dcNamespace, dcPodSelector, remotePodLister, remoteServiceLister)
 		if err != nil {
 			errs = append(errs, fmt.Errorf("can't calculate endpoints for %q ScyllaDBCluster %q Datacenter: %w", naming.ObjRef(sc), dc.Name, err))
 			continue
 		}
 
 		endpoints = append(endpoints, dcEndpoints...)
+
+		// Use the first detected IP family for the EndpointSlice
+		if detectedIPFamily == nil && dcIPFamily != nil {
+			detectedIPFamily = dcIPFamily
+		}
 	}
 
 	err = apimachineryutilerrors.NewAggregate(errs)
 	if err != nil {
 		return progressingConditions, nil, err
 	}
 
+	// Default to IPv4 if no IP family detected
+	addressType := discoveryv1.AddressTypeIPv4
+	if detectedIPFamily != nil && *detectedIPFamily == corev1.IPv6Protocol {
+		addressType = discoveryv1.AddressTypeIPv6
+	}
```

Comment:
If we moved `DetectEndpointsIPFamily` call outside of `calculateEndpointsForRemoteDCPods`, we could simplify deducing the address type and avoid repetition. Also, we could extract `IPFamilyToAddressType` to a helper.
```suggestion
		dcEndpoints, err := calculateEndpointsForRemoteDCPods(sc, clientBroadcastAddressType, dc, dcNamespace, dcPodSelector, remotePodLister, remoteServiceLister)
		if err != nil {
			errs = append(errs, fmt.Errorf("can't calculate endpoints for %q ScyllaDBCluster %q Datacenter: %w", naming.ObjRef(sc), dc.Name, err))
			continue
		}

		endpoints = append(endpoints, dcEndpoints...)
	}

	detectedIPFamily := helpers.DetectEndpointsIPFamily(endpoints)

	err = apimachineryutilerrors.NewAggregate(errs)
	if err != nil {
		return progressingConditions, nil, err
	}
```